File: app/src/main/java/com/tasomaniac/devwidget/widget/click/ClickHandlingModule.java
package com.tasomaniac.devwidget.widget.click;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class ClickHandlingModule {

    @Binds
    abstract Activity activity(ClickHandlingActivity activity);

    @Provides
    static ClickHandlingActivity.Input input(ClickHandlingActivity activity) {
        return activity.getIntent().getParcelableExtra(ClickHandlingActivity.EXTRA_INPUT);
    }
}

File: app/src/main/java/com/tasomaniac/devwidget/widget/click/ClickHandlingModule.kt
package com.tasomaniac.devwidget.widget.click

import android.app.Activity
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class ClickHandlingModule {

    @Binds
    abstract fun activity(activity: ClickHandlingActivity): Activity

    @Module
    companion object {

        @Provides
        @JvmStatic
        fun input(activity: ClickHandlingActivity) = activity.intent.input
    }
}
